src/Atmos/Model/AtmosModel.jl

```diff
@@ -682,23 +676,15 @@ function. Contributions from subcomponents are then assembled (pointwise).
     aux::Vars,
     t::Real,
 )
-    vec_pad = SVector(1, 1, 1)
-    tens_pad = SArray{Tuple{3, 3}}(ntuple(i -> 1, 9))
     tend = Flux{SecondOrder}()
-
     _args = (; state, aux, t, diffusive, hyperdiffusive)
-
     args = merge(_args, (precomputed = precompute(atmos, _args, tend),))
 
-    flux.ρ = Σfluxes(eq_tends(Mass(), atmos, tend), atmos, args) .* vec_pad
-    flux.ρu =
-        Σfluxes(eq_tends(Momentum(), atmos, tend), atmos, args) .* tens_pad
-
-    flux_second_order!(atmos.energy, flux, atmos, args)
-    flux_second_order!(atmos.moisture, flux, atmos, args)
-    flux_second_order!(atmos.precipitation, flux, atmos, args)
-    flux_second_order!(atmos.tracers, flux, atmos, args)
-    flux_second_order!(atmos.turbconv, flux, atmos, args)
+    foreach(prognostic_vars(atmos)) do prog
+        prog_flux = get_prog_state(flux, prog)
+        val = Σfluxes(prog, eq_tends(prog, atmos, tend), atmos, args)
+        setproperty!(prog_flux[1], prog_flux[2], val)
```

Comment:
```suggestion
        var, name = get_prog_state(flux, prog)
        val = Σfluxes(prog, eq_tends(prog, atmos, tend), atmos, args)
        setproperty!(var, name, val)
```

Reply:
Or even if we could combine it into a single function call?:
```suggestion
        Σfluxes!(flux, prog, eq_tends(prog, atmos, tend), atmos, args)
```
this might make it easier, as `Σfluxes!` can figure out the shape from the variable

Reply:
Given the scope of this PR, we can also change this later

Reply:
I missed the second suggestion here. I like the idea. So, `Σfluxes` would basically be defined as the contents of what is removed?

Reply:
I'll open a follow-up PR